packages/keyring-eth-trezor/src/trezor-keyring.ts

```diff
@@ -539,11 +539,6 @@ export class TrezorKeyring extends EventEmitter {
     // Reset HDKey if the path changes
     if (this.hdPath !== hdPath) {
       this.hdk = new HDKey();
```

Comment:
Curious: why the `HDKey` instance needs to be reset?

Reply:
Oh, maybe because `this.hdk` is initialized with the public key related to the selected path